pandas/core/arrays/sparse.py

```diff
@@ -1,6 +1,7 @@
 """
 SparseArray data structure
 """
+from collections.abc import Mapping
```

Comment:
If it'd be preferred, I could instead follow the pattern below:
```python
from collections import abc

...

if isinstance(mapper, abc.Mapping):
    ...
```

Reply:
I don't think we (as maintainers) have much of a preference on this.  My personal MO is that if you find yourself using it multiple times in the codebase, best to import directly into the namespace (fewer characters).

Reply:
From https://github.com/pandas-dev/pandas/pull/25957#pullrequestreview-221448898:

> we get to the point where we are doing from typing import Iterable

@WillAyd : Hmm...are we really sure we want to do that?  IIUC, `Iterable` is not immediately obvious as a "type" versus the actual `Iterable` object itself (e.g. from `collections.abc`).  IMO I would prefer that we namespace those to make it clear that they're types.

Reply:
Yea the convention for imports from typing are called out in MyPy documentation:

https://mypy.readthedocs.io/en/latest/getting_started.html#the-typing-module

Reply:
> My personal MO is that if you find yourself using it multiple times in the codebase, best to import directly into the namespace (fewer characters).

Agreed, that's my MO too.  The only reason I brought it up here is because these names can be a bit ambiguous, so it might be more readable and immediately obvious to see the `abc` prefix.  Aside from the `typing` conflicts there's also a conflict where we've named a custom class `Iterator`:

https://github.com/pandas-dev/pandas/blob/19626d2567487f8e698174de835c28c900784ed3/pandas/io/packers.py#L780

Will leave the PR as-is for now until more people weigh in.  I don't have a strong preference as long as we're consistent throughout the codebase.  The first thing I'd think upon seeing `Iterable` would be the `collections.abc` version but I'm likely biased from not having done much of anything with `typing`.

Reply:
Ah, I see...sigh...it's rather unfortunate that they decided have `Iterable` both refer to the object as well as the type.

Fair enough, then let's go with what Python says then.  However, I'm relatively certainly that we will be writing `abc.Iterable` many more times than we will be writing `Iterable` as a type hint (it's going to be more characters, I suspect 🙂)

Reply:
Oops, didn't refresh and see the link about the conventions before my last comment; `abc` prefix it is then.